### PR TITLE
Add test documentation and expand CLI test coverage

### PR DESCRIPTION
**Pull request by kaladron**
Implement standard exit codes and expand CLI test suite with better documentation.

- Implement standard POSIX/GNU tar exit codes in `src/uu/tar/src/tar.rs`:
  * Map `clap` argument errors (unknown options, missing arguments) to usage error code (64)
  * Update fatal errors (missing files, empty archives, conflicting options) to return exit code 2
  * Add explicit `conflicts_with` between `-c` and `-x` operations to ensure clear error reporting
- Add `tests/README.md` documenting the testing philosophy: separating application (CLI interface/UX) from library (archive format processing) concerns
- Update `README.md` with a "Testing" section linking to the new documentation
- Reorganize and expand the test suite in `tests/by-util/test_tar.rs` (expanded from 9 to 26 tests):
  * Improve all existing tests with `code_is()` checks and `stderr_contains()` pattern matching
  * Add `no_output()` assertions for successful silent operations
  * Add sanity checks for archive sizes using `TAR_BLOCK_SIZE`
  * Add 17 new CLI-focused tests covering:
    * basic CLI: conflicting operations, missing operation specification
    * Create: directory archiving, verbose output format, empty archive handling, nonexistent file errors
    * Extract: verbose output, directory structure preservation, overwriting default behavior
    * Error handling: permission denied (Unix), corrupted archives, dash-prefixed filenames
    * CLI parsing: mixed short/long options, option order variations
    * Edge cases: filenames with spaces, large file counts (100 files)

Tested:
All 26 tests pass.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Testing the tar Application
+
+This directory contains tests for the `tar` command-line utility.
+
+## Philosophy
+
+The `tar` utility is built on top of the `tar-rs` library. Because of this, we split our testing into two distinct areas:
+
+1.  **The Library (`tar-rs`)**: This is where we test the nitty-gritty details of the tar format. If you want to verify that permissions are preserved correctly, that long paths are handled according to the UStar spec, or that unicode filenames are encoded properly, those tests belong in `tar-rs/tests/`.
```
Comment (cakebaker) [inline tests/README.md]: I don't know if it's a planned refactoring, but currently there is no `tar-rs/tests` folder.

File: tests/by-util/test_tar.rs [hunk 2]
```diff
@@ -49,14 +77,69 @@ fn test_create_multiple_files() {
 
     at.write("file1.txt", "content1");
     at.write("file2.txt", "content2");
+    at.write("file3.txt", "content3");
 
-    ucmd.args(&["-cf", "archive.tar", "file1.txt", "file2.txt"])
-        .succeeds();
+    ucmd.args(&["-cf", "archive.tar", "file1.txt", "file2.txt", "file3.txt"])
+        .succeeds()
+        .no_stderr();
```
Comment (cakebaker) [inline tests/by-util/test_tar.rs]:
```suggestion
        .no_output();
```

File: tests/by-util/test_tar.rs [hunk 3]
```diff
@@ -112,5 +217,478 @@ fn test_extract_nonexistent_archive() {
...
+    at.write("testdir/subdir/file2.txt", "content2");
+
+    // Create archive
+    ucmd.args(&["-cf", "archive.tar", "testdir"]).succeeds();
+
+    // Remove directory contents and directory itself
+    at.remove("testdir/subdir/file2.txt");
+    at.remove("testdir/file1.txt");
+    std::fs::remove_dir(at.plus("testdir/subdir")).unwrap();
+    std::fs::remove_dir(at.plus("testdir")).unwrap();
```
[the remark this answers is not in the document]
Comment (kaladron) [reply tests/by-util/test_tar.rs]: I'm still learning, so sorry for the silly question.  Why do we prefer the at. method  instead of remove_dir?

File: tests/by-util/test_tar.rs [hunk 4]
```diff
@@ -30,17 +36,39 @@ fn test_version() {
...
+        .fails()
+        .code_is(2);
+}
+
+#[test]
+fn test_no_operation_specified() {
+    new_ucmd!()
+        .args(&["-f", "archive.tar"])
+        .fails()
+        .code_is(1) // TODO: align with GNU tar by returning exit code 64
```
[the remark this answers is not in the document]
Comment (kaladron) [reply tests/by-util/test_tar.rs]: I'll double check this (and below) but I had tested these by hand for comparison and I think these were the ones returning 64.  The GNU tar documentation only document exit 0,1, and 2.

File: tests/by-util/test_tar.rs [hunk 5]
```diff
@@ -112,5 +217,478 @@ fn test_extract_nonexistent_archive() {
...
+    let result = ucmd
+        .args(&["-cvf", "archive.tar", "file1.txt", "file2.txt"])
+        .succeeds();
+
+    let stdout = result.stdout_str();
+
+    // Verify verbose output contains filenames
+    assert!(stdout.contains("file1.txt"));
+    assert!(stdout.contains("file2.txt"));
+}
```
Comment (cakebaker) [inline tests/by-util/test_tar.rs]: This test is more or less the same as `test_create_verbose`, with the only difference that this test uses two files.
```suggestion
```

**Comment by sylvestre** [comment]
ping ? :)

**Comment by sylvestre** [comment]
welcome back! :)

File: tests/by-util/test_tar.rs [hunk 6]
```diff
@@ -49,14 +77,69 @@ fn test_create_multiple_files() {
...
+
+    assert!(at.file_exists("archive.tar"));
+}
+
+#[test]
+fn test_create_empty_archive_fails() {
+    new_ucmd!()
+        .args(&["-cf", "archive.tar"])
+        .fails()
+        .code_is(1) // TODO: propagate usage exit code 64 once empty archive handling is fixed
```
[the remark this answers is not in the document]
Comment (kaladron) [reply tests/by-util/test_tar.rs]: Not only is this 2 as you suggested, it's no longer a TODO =)

File: tests/by-util/test_tar.rs [hunk 7]
```diff
@@ -112,5 +217,478 @@ fn test_extract_nonexistent_archive() {
...
+
+    // Verify empty files exist and are still empty
+    assert!(at.file_exists("empty1.txt"));
+    assert!(at.file_exists("empty2.txt"));
+    assert_eq!(at.read("empty1.txt"), "");
+    assert_eq!(at.read("empty2.txt"), "");
+}
+
+#[test]
+fn test_roundtrip_special_characters_in_names() {
```
[the remark this answers is not in the document]
Comment (kaladron) [reply tests/by-util/test_tar.rs]: This should probably go in tar-rs anyway, migrating this to there. (Opened an issue for me to track this)

**Comment by kaladron** [comment]
Okay, so apparently 'cargo clippy' isn't enough.  Sorry about that!

File: src/uu/tar/src/tar.rs [hunk 8]
```diff
@@ -31,14 +31,39 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
+        }
+    };
 
     let verbose = matches.get_flag("verbose");
 
     // Handle extract operation
     if matches.get_flag("extract") {
         let archive_path = matches.get_one::<PathBuf>("file").ok_or_else(|| {
-            uucore::error::USimpleError::new(1, "option requires an argument -- 'f'")
+            uucore::error::USimpleError::new(64, "option requires an argument -- 'f'")
```
Comment (sylvestre) [inline src/uu/tar/src/tar.rs]: are you sure ?

tar -f somefile ; echo "GNU exit: $?"

returns 2 as exit code
